bigquery/samples/tests/conftest.py

```diff
@@ -34,6 +34,15 @@ def random_table_id(client, dataset_id):
     return "{}.{}".format(dataset_id, random_table_id)
 
 
+@pytest.fixture
+def random_dataset_id(client):
+    now = datetime.datetime.now()
+    random_dataset_id = "example_dataset_{}_{}".format(
+        now.strftime("%Y%m%d%H%M%S"), uuid.uuid4().hex[:8]
+    )
+    return "{}.{}".format(client.project, random_dataset_id)
```

Comment:
You'll probably notice that a dataset is created but not deleted when you run these tests. Our tests should clean up after themselves. In this case, we should yield the ID, and then afterwards delete the dataset.

Set both `delete_contents=True` and `not_found_ok=True` when you call [`delete_dataset`](https://googleapis.github.io/google-cloud-python/latest/bigquery/generated/google.cloud.bigquery.client.Client.html#google.cloud.bigquery.client.Client.delete_dataset) so that this fixture can be used in samples that create tables within the dataset or don't create a dataset at all (such as if there was a test failure).

Reply:
Changed `return` to `yield` and added the cleanup. Also added the `delete_contents=True` and `not_found_ok=True` as directed. 

In addition, applied `not_found_ok=True` to the dataset_id fixture for consistency. Please let me know if it should be removed!

Reply:
Great, thank you. That will come in handy for the "delete dataset" snippets.